webapp/channels/src/utils/popouts/browser_popouts.test.ts

```diff
@@ -0,0 +1,595 @@
+
+            browserPopouts.setupBrowserPopout('/_popout/test');
+
+            const callArgs = mockWindowOpen.mock.calls[0];
+
+            // These are be based on where the window is opened relative to the screen
+            expect(callArgs[2]).toContain('left=900');
+            expect(callArgs[2]).toContain('top=150');
+            expect(callArgs[2]).toContain('width=800');
+            expect(callArgs[2]).toContain('height=700');
+        });
+
+        it('should handle multiple popout windows independently', () => {
+            const mockPopoutWindow1 = {
+                closed: false,
+                postMessage: jest.fn(),
+            } as unknown as Window;
+
+            const mockPopoutWindow2 = {
+                closed: false,
+                postMessage: jest.fn(),
+            } as unknown as Window;
+
+            mockWindowOpen.mockReturnValueOnce(mockPopoutWindow1).mockReturnValueOnce(mockPopoutWindow2);
+
+            const result1 = browserPopouts.setupBrowserPopout('/_popout/test1');
+            const result2 = browserPopouts.setupBrowserPopout('/_popout/test2');
+
+            const listener1 = jest.fn();
+            const listener2 = jest.fn();
+
+            if (result1.message) {
+                result1.message(listener1);
+            }
+            if (result2.message) {
+                result2.message(listener2);
+            }
+
+            const messageEvent1 = new MessageEvent('message', {
+                data: {
+                    channel: 'channel1',
+                    args: [],
+                },
+                origin: 'https://example.com',
+                source: mockPopoutWindow1,
+            });
+
+            const messageEvent2 = new MessageEvent('message', {
+                data: {
+                    channel: 'channel2',
+                    args: [],
+                },
+                origin: 'https://example.com',
+                source: mockPopoutWindow2,
+            });
+
+            messageListeners.forEach((listener) => {
+                listener(messageEvent1);
+                listener(messageEvent2);
+            });
+
+            expect(listener1).toHaveBeenCalledWith('channel1');
+            expect(listener2).toHaveBeenCalledWith('channel2');
+            expect(listener1).not.toHaveBeenCalledWith('channel2');
+            expect(listener2).not.toHaveBeenCalledWith('channel1');
+        });
+    });
+
+    describe('send', () => {
+        it('should handle multiple arguments', () => {
+            const mockPopoutWindow = {
+                closed: false,
+                postMessage: jest.fn(),
+            } as unknown as Window;
+
+            mockWindowOpen.mockReturnValue(mockPopoutWindow);
+
+            const result = browserPopouts.setupBrowserPopout('/_popout/test');
+            if (result.send) {
+                result.send('channel', 1, true, {nested: 'object'}, ['array']);
+            }
```

Comment:
This is a test, so result.send should exist. I would argue that we should use the `!` operator, and let it crash if send is not there.